pkg/util/hash.go

```diff
@@ -0,0 +1,25 @@
+// © Broadcom. All Rights Reserved.
+// The term “Broadcom” refers to Broadcom Inc. and/or its subsidiaries.
+// SPDX-License-Identifier: Apache-2.0
+
+package util
+
+import (
+	"crypto/sha1" //nolint:gosec // used for creating safe names
+	"encoding/hex"
+	"io"
+	"strings"
+)
+
+// SHA1Sum17 returns the first 17 characters of the base64-encoded, SHA1
+// sum created from the provided string.
+func SHA1Sum17(s string) string {
+	h := sha1.New() //nolint:gosec // used for creating safe names
```

Comment:
Go 1.24 is supposed to remove sha1 support.. should we use sha256 or md5 here instead?

Reply:
Hmm, that's not good as the image registry uses Sha1 everywhere for VMI IDs. What is the plan for that?

Reply:
Ah ok, that makes more sense. iirc "sha1 is being removed in 1.24" was raised in govmomi slack ~1 year ago, but it's just sha-1 signature support going away, not the whole package. Sorry for the confusion.